client/controller/shared.js

```diff
@@ -35,25 +37,86 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/**
+ * Loads a a css stylesheet into the page
+ * @param {string} cssUrl - a url to a css resource to be inserted into the page
+ * @param {Function} callback - a callback function to be called when the CSS has been loaded (after 500ms have passed).
+ */
+function loadCSS( cssUrl, callback = noop ) {
+	const link = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: cssUrl,
+	} );
+
+	const onload = () => {
+		if ( 'onload' in link ) {
+			link.onload = null;
+		}
+		callback( null, link );
+	};
+
+	if ( 'onload' in link ) {
+		link.onload = onload;
+	} else {
+		setTimeout( onload, 500 );
```

Comment:
It's not clear to me why we have this arbitrary timeout.

Reply:
https://github.com/Automattic/wp-calypso/pull/17988/files#r141029175

Reply:
I wished we added an inline comment explaining the reason for that timeout 😞 .